pkg/dependencies/dependencies.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dependencies
+
+import (
+	"fmt"
+	"github.com/a8m/envsubst/parse"
+	"github.com/okteto/okteto/pkg/env"
+	giturls "github.com/whilp/git-urls"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+)
+
+// ManifestDependencies represents the map of dependencies at a manifest
+type ManifestDependencies map[string]*Dependency
```

Comment:
Nit: not blocker.

`dependencies.ManifestDependencies` is a bit repetitive, maybe we can call it `dependencies.ManifestSection`? Also, taking into account my previous comment, it could be `deps.ManifestDependencies` or `dependencies.ManifestDeps`

Reply:
`dependency.Manifest` ?

Reply:
+1 on going with the `ManifestSection`, since it's how we are referring through the meetings and issues created to cover this functionality

Reply:
I agree the name is long and we could try find a shorter version but, I am afrai I don't agree with the proposed names. This type represent the **dependencies** in the manifest, not the manifest. The singular type is `deps.Depencency` which represent the single dependency. How can the array of dependencies be called `deps.ManifestSection`? I would rather call it `deps.Dependencies`.

cc @ifbyol @teresaromero @jLopezbarb

Reply:
The package already give it the context of being dependencies. The manifestSection is a list, that's why it would be a list.

The thing is, the package name should provide context about the type used. For example, imagine you have 3 packages: `dependencies`, `external-resources` and `endpoints`. You want to create a `Deployer` for each of them to deploy each type as part of the deploy. You don't need to call them `DependenciesDeployer`, `ExternalResourcesDeployer` and `EndpointsDeployer`. As the package will be used, you can call them just `Deployer` on each package and then, they would be used like: `dependencies.Deployer`, `external-resources.Deployer` and `endpoints.Deployer`

`deps.Dependencies` would be repetitive from a client perspective and doesn't provide the context that refer to the dependencies defined in the manifest. It might be just an internal model of dependencies and not the one extracted from unmarshaling the manifest. This [blog post](https://go.dev/blog/package-names) talks a bit about package names, package content name and its relationship. 

With this, I'm not saying we should use our suggestions. Just giving some rationale on why the proposed names. In the struct name, I think it is more important to indicate Manifest than Dependencies. Dependencies context is already gotten from the package name while nothing is giving the context of the Manifest. Take into account that this type is implementing an `UnmarshalYAML` function because it is tightly coupled to the manifest parsing.

Reply:
@ifbyol I think it depends on the point of view on how you see the dependencies. I personally like to view them as a concept/feature related to a deployment, so they could come from a CLI flag, for example. `okteto deploy --repository XXX --dependency YYY --dependency ZZZ` in the future. In that case for me they would still be stored as `deps.Dependencies`. The fact that we deserialize dependencies from a manifest, for me it doesn't imply that we call them `ManifestSection` for example.

Reply:
Changed in 946e7495

Reply:
Then, I think the abstraction wouldn't be correct and we would need to split between the internal model needed by the CLI, and the one from the manifest and command flags. In the same way an internal model shouldn't be coupled to an API response. A change in the API would imply to change a lot of internal code to fit the new response when everything you need is an adapter between the API and the model.

I would find odd that a struct in charge of receiving the flags from the command would have a function to unmarshall it from a yaml file. Probably the way a dependency can be specified from the flag and the manifest is different and they should be separated. That's why I think it is important the context of the manifest in the name.

Anyway, as you mentioned, it depends on the point of view and personal preferences, I don't have a strong opinion about it